caddyconfig/httpcaddyfile/tlsapp.go

```diff
@@ -523,7 +523,7 @@ func consolidateAutomationPolicies(aps []*caddytls.AutomationPolicy) []*caddytls
 					// '*.com', which might be different (yes we've seen this happen)
 					if automationPolicyShadows(i, aps) >= j {
 						aps = append(aps[:i], aps[i+1:]...)
-						i--
+						i = max(0, i-1)
```

Comment:
For reference, this decrement was added in https://github.com/caddyserver/caddy/commit/7846bc1e06b2de97906cd562d16db4b2aafbd74b.